app/cp_nigeria/views.py

```diff
@@ -1505,14 +1505,21 @@ def save_graph_to_db(request, proj_id):
             graph_id = "stacked_timeseries"
         attr_name = f"{graph_id}_graph"
         image_url = request.POST.get("image_url")
-        report_qs = ImplementationPlanContent.objects.filter(simulation=project.scenario.simulation)
-        if report_qs.exists():
-            report_content = report_qs.first()
-            setattr(report_content, attr_name, image_url)
-            report_content.save()
-            return JsonResponse({"status": "saved " + attr_name + " to db"})
-        else:
-            return JsonResponse({"status": "db object could not be found"})
+
+        try:
+            with transaction.atomic():
+                report_qs = ImplementationPlanContent.objects.select_for_update().filter(
+                    simulation=project.scenario.simulation
+                )
+                if report_qs.exists():
+                    report_content = report_qs.first()
+                    setattr(report_content, attr_name, image_url)
+                    report_content.save()
+                    return JsonResponse({"status": "success", "message": "Saved " + attr_name + " to database"})
```

Comment:
I prefer having only one `return` statement in a function, I just store the returned value in a variable like `answer` and I write `return answer` only once at the end of the function, this helps by debugging, you don't need to set traces before each of your return statements